Fix error when safe yaml is only partially loaded

maxjacobson opened the pull request:
This should fix #2218 and #2215

The error was introduced here: https://github.com/bbatsov/rubocop/commit/2c32a61f5bf144a3c824c4287398b317a0f51096 and kind of here: https://github.com/sferik/rails_admin/commit/c1a495edf40124dbbb391acb4e9240f2ac15f8a9 and can be resolved by following the example of that second commit

maxjacobson commented:
Tried three approaches (https://github.com/maxjacobson/rubocop/commit/d22f633d92a3c2d5663f8c130d13fabda8743000, https://github.com/maxjacobson/rubocop/commit/12849814f33b746e52d7eb2a82278aa7fb389784, https://github.com/maxjacobson/rubocop/commit/8b9a2b39913044623012636fa3714cbc72bb3a05) and I think the last one is best, so I'm squashing.

maxjacobson commented on spec/rubocop/config_loader_spec.rb:
```diff
@@ -292,6 +292,8 @@
     end
 
+    # TODO: figure out why these specs print a green dot even when they fail
+    # (probably related to the process shenanigans)
```
this is a bit concerning, I think, but I have NO idea how to resolve it :grimacing:

bbatsov commented:
//cc @eitoball

Those safe_yml issues are killing me...

maxjacobson commented:
The error kind of makes sense in retrospect / after reading the safe yaml readme. They encourage you to choose whether you want to monkey patch YAML or not, so rubocop can't assume that the monkey patch occurred

maxjacobson commented:
Pushed an update to avoid duplicating too much code, and to assert on the successful execution of the subprocess (otherwise failing tests are green)

rud commented on spec/rubocop/config_loader_spec.rb:
```diff
@@ -300,25 +300,27 @@
-
+      context 'when it is fully required' do
+        it 'de-serializes Regexp class' do
+          in_its_own_process do
```
That's a pretty neat helper for clearing up the intention.

maxjacobson replied:
I'm about to update it to, I think, make it even more intentional :whoa:

maxjacobson commented:
Squashed again, but if you're interested in the changes since the last commit, they were: https://github.com/maxjacobson/rubocop/commit/eebc66309bcc5756313f89519aa274330d0ba2e9 and https://github.com/maxjacobson/rubocop/commit/78d65139605d4ff3953606d34c589071ce23bbd4

maxjacobson commented:
One more :blush: https://github.com/maxjacobson/rubocop/commit/5d6fe78ea3a9c79b151fe8002f6015dc9b593f32

bbatsov commented:
:+1:
